src/FastRouteRouter.php

```diff
@@ -114,12 +159,14 @@ public function match(Request $request)
         // Inject any pending routes
         $this->injectRoutes();
 
+        $dispatchData = $this->getDispatchData();
+
         $path       = $request->getUri()->getPath();
         $method     = $request->getMethod();
-        $dispatcher = $this->getDispatcher($this->router->getData());
+        $dispatcher = $this->getDispatcher($dispatchData);
         $result     = $dispatcher->dispatch($method, $path);
 
-        if ($result[0] != Dispatcher::FOUND) {
+        if ($result[0] !== Dispatcher::FOUND) {
```

Comment:
@Ocramius 
I believed i asked the author for a Result object a long time ago, but i believe he wanted to keep that part of the code as simpler as possible, without interfaces, etc etc...